File: Data Structures & Algorithms/is-anagram/submission-9.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if (len(s) != len(t)):
            return False
        hs = {}
        ht = {}
        for i in s:
            if i in hs:
                hs[i] += 1
            else:
                hs[i] = 1
        for i in t:
            if i in ht:
                ht[i] += 1
            else:
                ht[i] = 1
        return hs == ht

File: Data Structures & Algorithms/anagram-groups/submission-9.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> map;
        for (const auto& s : strs) {
            string key = s;
            sort(key.begin(), key.end());
            map[key].push_back(s);
        }
        vector<vector<string>> vec;
        for (const auto& pair : map) {
            vec.push_back(pair.second);
        }
        return vec;
    }
};
